salt/modules/win_service.py

```diff
@@ -100,6 +98,50 @@ def __virtual__():
     return __virtualname__
 
 
+def _status_wait(service_name, end_time, service_states):
+    '''
+    Helper function that will wait for the status of the service to match the
+    provided status before an end time expires. Used for service stop and start
+
+    .. versionadded:: 2017.7.9, 2018.3.4
+
+    Args:
+        service_name (str):
+            The name of the service
+
+        end_time (float):
+            A future time. e.g. time.time() + 10
+
+        service_states (list):
+            Services statuses to wait for as returned by info()
+
+    Returns:
+        dict: A dictionary containing information about the service.
+
+    :codeauthor: Damon Atkins <https://github.com/damon-atkins>
+    '''
+    info_results = info(service_name)
+
+    while info_results['Status'] in service_states and time.time() < end_time:
+        # From Microsoft: Do not wait longer than the wait hint. A good interval
+        # is one-tenth of the wait hint but not less than 1 second and not more
+        # than 10 seconds.
+        # https://docs.microsoft.com/en-us/windows/desktop/services/starting-a-service
+        # https://docs.microsoft.com/en-us/windows/desktop/services/stopping-a-service
+        wait_time = info_results['Status_WaitHint'] / 1000
```

Comment:
Looks like I got this code slightly wrong.
Should not have the / 1000   see the else which should do this. This way we cover the off chance wait hint is 0.

``
else:
           wait_time = wait_time / 10000  # convert ms to seconds, and then divide by 10.
``